src/main/kotlin/net/minecrell/pluginyml/PlatformPlugin.kt

```diff
@@ -56,9 +60,9 @@ abstract class PlatformPlugin<T : PluginDescription>(private val platformName: S
                 }
             }
 
-            plugins.withType<JavaPlugin> {
-                tasks.named<AbstractCopyTask>("processResources") {
-                    from(generateTask)
+            if (plugins.hasPlugin(JavaPlugin::class)) {
```

Comment:
Sorry, I know it's confusing but this is not equivalent to `withType`. :)

`withType` registers an action that is called sometime later, when the JavaPlugin was applied. So it works even if you apply the JavaPlugin _after_ plugin-yml. Your if statement on the other hand works only if the JavaPlugin was applied _before_ plugin-yml.

Actually, this breaks even my most trivial plugin-yml test project, with

```kotlin
plugins {
    id("java")
    id("net.minecrell.plugin-yml.bukkit") version "0.5.0-SNAPSHOT"
    id("net.minecrell.plugin-yml.bungee") version "0.5.0-SNAPSHOT"
    id("net.minecrell.plugin-yml.nukkit") version "0.5.0-SNAPSHOT"
}
```

because apparently Gradle registers the plugins in a different order for some reason:

```
Build operation 'Apply plugin net.minecrell.plugin-yml.bukkit to root project 'pluginymltest'' started
Build operation 'Apply plugin net.minecrell.plugin-yml.bungee to root project 'pluginymltest'' started
Build operation 'Apply plugin net.minecrell.plugin-yml.nukkit to root project 'pluginymltest'' started
Build operation 'Apply plugin org.gradle.java to root project 'pluginymltest'' started
```

Reply:
Oops